### DISCUSSION TITLE
Treehouse can update business logic?

### DISCUSSION THREAD

**Discussion by jeonhoeun** [cashapp/redwood#1449] [Q&A]
https://www.youtube.com/watch?v=G4LK_euTadU

Hello,

upper video says  TreeHouse = zipline+redwood
and TreeHouse is a library for UI what can update without app update at store(google play store and apple app store )

I have a question.

Does TreeHouse can update business logic without update at store?

or TreeHouse can update just UI? ( something like, developer mistakes word  or icon image .. etc )

**Comment by JakeWharton** [comment]
If the logic is contained within the presentation code written in Compose, or any of the behavior bundled with the Compose logic, then yes. If the logic is implemented by the "host" (i.e., the native app) and accessed through a Zipline service then those are only able to be updated by an app release.
